File: tp/Translator.java
package tp;

import io.vertx.core.AbstractVerticle;
import io.vertx.core.Handler;
import io.vertx.core.buffer.Buffer;
import io.vertx.core.eventbus.EventBus;
import io.vertx.core.eventbus.Message;

import java.io.IOException;

public class Translator extends AbstractVerticle {

    public static void main(String[] args) {
	Runner.runClusteredExample(Translator.class);
    }

        @Override
	public void start() throws Exception {
	    EventBus eb = vertx.eventBus();

	    eb.consumer("translator", (Handler<Message<Buffer>>) message -> {
		    String p = message.body();

		    System.out.println(p);
		    Translator translate = Translator.getInstance();
		    String text = translate.translate("Hello!", Language.ENGLISH, Language.ROMANIAN);
		    System.out.println(text);
	    });

	}
}

File: tp/Translator.js
/* "use strict" */

var eb = vertx.eventBus();
var httpHandler = vertx.createHttpClient();

console.log("vertx: %j", vertx);
console.log("httpHandler: %j", httpHandler);
eb.consumer("translator", function (message) {

    console.log(message.body());
    httpHandler.getAbs("https://www.google.com", function (res) {
        console.log("Got response " + res.statusCode());
        res.bodyHandler(function (body) {
            console.log("Got data " + body);
        });
    });
});